.github/actions/fullnode-sync/fullnode_sync.py

```diff
@@ -0,0 +1,267 @@
+  """Spawns the fullnode"""
+  # Display the fullnode setup
+  print("Starting the fullnode using branch: {branch}, for network: {network}, " \
+        "with bootstrapping mode: {bootstrapping_mode} and continuous syncing " \
+        "mode: {continuous_syncing_mode}!".format(branch=branch, network=network,
+                                                  bootstrapping_mode=bootstrapping_mode,
+                                                  continuous_syncing_mode=continuous_syncing_mode))
+
+  # Display the fullnode config
+  with open(FULLNODE_CONFIG_NAME) as file:
+    fullnode_config = yaml.safe_load(file)
+  if fullnode_config is None:
+    print_error_and_exit("Exiting! Failed to load the fullnode config template at {template_path}!".format(template_path=FULLNODE_CONFIG_NAME))
+  print("Starting the fullnode using the config: {fullnode_config}".format(fullnode_config=fullnode_config))
+
+  # Start the fullnode
+  node_log_file = open(node_log_file_path, "w")
+  process_handle = subprocess.Popen(["cargo", "run", "-p", "aptos-node", "--release", "--", "-f", FULLNODE_CONFIG_NAME], stdout=node_log_file, stderr=node_log_file)
+
+  # Return the process handle
+  return process_handle
+
+
+def setup_fullnode_config(bootstrapping_mode, continuous_syncing_mode, data_dir_file_path):
+  """Initializes and configures the fullnode config file"""
+  # Copy the node config template to the working directory
+  if not os.path.exists(FULLNODE_CONFIG_TEMPLATE_PATH):
+    print_error_and_exit("Exiting! The fullnode config template wasn't found: {template_path}!".format(template_path=FULLNODE_CONFIG_TEMPLATE_PATH))
+  subprocess.run(["cp", FULLNODE_CONFIG_TEMPLATE_PATH, FULLNODE_CONFIG_NAME])
+
+  # Update the data_dir in the node config template
+  with open(FULLNODE_CONFIG_NAME) as file:
+    fullnode_config = yaml.safe_load(file)
+  if fullnode_config is None:
+    print_error_and_exit("Exiting! Failed to load the fullnode config template at {template_path}!".format(template_path=FULLNODE_CONFIG_TEMPLATE_PATH))
+  fullnode_config['base']['data_dir'] = data_dir_file_path
+
+  # Add the state sync configurations to the config template
+  state_sync_driver_config = {"bootstrapping_mode": bootstrapping_mode, "continuous_syncing_mode": continuous_syncing_mode}
+  data_streaming_service_config = {"max_concurrent_requests": 10}
+  fullnode_config['state_sync'] = {"state_sync_driver": state_sync_driver_config, "data_streaming_service":data_streaming_service_config}
+
+  # Write the config file back to disk
+  with open(FULLNODE_CONFIG_NAME, "w") as file:
+    yaml.dump(fullnode_config, file)
+
+
+def get_genesis_and_waypoint(network):
+  """Clones the genesis blob and waypoint to the current working directory"""
+  # Clone the aptos-networks repository at the same place as aptos-core
+  current_working_directory = os.getcwd()
+  os.chdir("..")
+  subprocess.run(["git", "clone", "https://github.com/aptos-labs/aptos-networks"])
```

Comment:
nit: instead of git clone, you can prob just get it from the raw paths here: e.g. https://raw.githubusercontent.com/aptos-labs/aptos-networks/main/mainnet/waypoint.txt
That way, you can get around having to do git magic and chdir ..

Reply:
Nice! So much simpler 😄